### ISSUE TITLE
[REVIEW]: MTEX2Gmsh: a tool for generating 2D meshes from EBSD data

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2094]
**Submitting author:** @DorianDepriester (<a href="http://orcid.org/0000-0002-2881-8942">Dorian Depriester</a>)
**Repository:** <a href="https://github.com/DorianDepriester/mtex2Gmsh" target ="_blank">https://github.com/DorianDepriester/mtex2Gmsh</a>
**Version:** v1.0
**Editor:** @meg-simula
**Reviewer:** @streeve, @ralfHielscher
**Archive:** Pending

## Status

[![status](https://joss.theoj.org/papers/aef990e2d5098db30344a068f978ddbd/status.svg)](https://joss.theoj.org/papers/aef990e2d5098db30344a068f978ddbd)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/aef990e2d5098db30344a068f978ddbd"><img src="https://joss.theoj.org/papers/aef990e2d5098db30344a068f978ddbd/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/aef990e2d5098db30344a068f978ddbd/status.svg)](https://joss.theoj.org/papers/aef990e2d5098db30344a068f978ddbd)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/DorianDepriester/mtex2Gmsh" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@streeve & @ralfHielscher, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @meg-simula know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [ ] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/DorianDepriester/mtex2Gmsh">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Contribution and authorship:** Has the submitting author (@DorianDepriester) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [ ] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/DorianDepriester/mtex2Gmsh">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Contribution and authorship:** Has the submitting author (@DorianDepriester) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @streeve, @ralfHielscher it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1088/0965-0393/24/5/055014 is OK
- 10.1016/j.actamat.2016.02.001 is OK
- 10.1016/0956-7151(94)90452-9 is OK
- 10.1364/AO.41.006815 is OK
- 10.1016/j.actamat.2016.10.030 is OK
- 10.1002/nme.2579 is OK
- 10.1016/j.ultramic.2011.08.002 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02094/joss.02094/10.21105.joss.02094.pdf)

**Comment by meg-simula** [comment]
@streeve @ralfHielscher Thanks again for agreeing to review. Reviewer checklists have been generated for each you above - take a look and dig in at will.

**Comment by meg-simula** [comment]
@streeve Thanks for your comments on this submission, looks like the review is progressing nicely. 

@ralfHielscher Have you had a chance to start looking at this?

**Comment by DorianDepriester** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02094/joss.02094/10.21105.joss.02094.pdf)

**Comment by streeve** [comment]
@meg-simula @DorianDepriester 

I've completed the checklist and closed all the issues with my suggestions. This is a very nice code, combining the strengths of underlying tools. I'm no expert in this type of simulation, but the docs and examples make it clear what I could do with it. 

One more note: I did not test the Abaqus portion (I do not have access)

**Comment by meg-simula** [comment]
Thanks @streeve for your prompt review and @DorianDepriester for the prompt response!

**Comment by meg-simula** [comment]
@ralfHielscher :wave:  We are waiting on your review to proceed with this submission.

**Comment by arfon** [comment]
Dear authors and reviewers

We wanted to notify you that in light of the current COVID-19 pandemic, JOSS has decided to suspend submission of new manuscripts and to handle existing manuscripts (such as this one) on a "best efforts basis". We understand that you may need to attend to more pressing issues than completing a review or updating a repository in response to a review. If this is the case, a quick note indicating that you need to put a "pause" on your involvement with a review would be appreciated but is not required.

Thanks in advance for your understanding.

_Arfon Smith, Editor in Chief, on behalf of the JOSS editorial team._

**Comment by arfon** [comment]
:wave: @ralfHielscher, just a friendly check-in to see how things are going with your review?

**Comment by arfon** [comment]
👋 @ralfHielscher - today we [reopened JOSS](https://blog.joss.theoj.org/2020/05/reopening-joss) for new submissions and are checking in on our existing reviews. Do you think you might be able to wrap up your review in the next 2-3 weeks?

**Comment by meg-simula** [comment]
@ralfHielscher :wave: We are still waiting on your review to proceed with this submission. Could you give us an update?

**Comment by meg-simula** [comment]
Sorry for the delay on this review @DorianDepriester . I will unassign @ralfHielscher and find an alternative reviewer to get this process completed.

**Comment by DorianDepriester** [comment]
Hello there,
Just to mention that @ralfHielscher has kindly reviewed the submission.
https://github.com/DorianDepriester/MTEX2Gmsh/issues/5#issue-648501620
I am doing my best to promptly answer each bullet point.

**Comment by meg-simula** [comment]
Ok, thanks @ralfhielscher and @doriandepriester!

**Comment by meg-simula** [comment]
Just checking in here - all the boxes have been checked it seems - what are your thoughts on the current state of the submission @DorianDepriester @ralfHielscher

**Comment by DorianDepriester** [comment]
I have answered all the point raised by @ralfHielscher in the issue [mentioned above](https://github.com/DorianDepriester/MTEX2Gmsh/issues/5#issue-648501620). He seems satisfied for he has closed the issue.

**Comment by meg-simula** [comment]
Ok, great! I will proceed with this submission once I'm back from vacation.

**Comment by meg-simula** [comment]
@whedon check references

**Comment by meg-simula** [comment]
@whedon generate pdf

**Comment by meg-simula** [comment]
@DorianDepriester Thanks for your patience with this submission. 

I've gone through the paper, and ask that you please consider the following suggestions:

- [ ] Include a clear Statement of Need that illustrates the research purpose of the software.
- [ ] Mention (if applicable) a representative set of past or ongoing research projects using the software and recent scholarly publications enabled by it.
- [ ] Acknowledge any financial support.
- [ ] This sentence is hard to understand: "As part of characterization techniques, Orientation Imaging Microscopy (OIM), usually made from Electron Backscatter Diffraction (EBSD) data, is now widely used in the literature for investigating the grain morphology and local crystal orientations in crystalline materials." Rephrase. 
- [ ] Similarly, rephrase: "Nevertheless, because of the regular structure, the GBs can not be smoothly described"
- [ ] Replace  "theory -based" by "theory-based".

**Comment by meg-simula** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1088/0965-0393/24/5/055014 is OK
- 10.1016/j.actamat.2016.02.001 is OK
- 10.1016/0956-7151(94)90452-9 is OK
- 10.1364/AO.41.006815 is OK
- 10.1016/j.actamat.2016.10.030 is OK
- 10.1002/nme.2579 is OK
- 10.1016/j.ultramic.2011.08.002 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by DorianDepriester** [comment]
@meg-simula, here are the answers to all the points you have raised:

- [x] A clear Statement of Need has been added (see https://github.com/DorianDepriester/MTEX2Gmsh/commit/afd5ebbe34fb63173765c18be54a48dadc0aa6ac )
- [x] There is no representative set of past or ongoing research projects using the software and nor recent scholarly publications enabled by it.
- [x] This project is not financially supported, thus there is no need for any acknowledgement
- [x] The sentence has been rephrased (see https://github.com/DorianDepriester/MTEX2Gmsh/commit/d251c90a6896240a192db27394398ea974ed55b7)
- [x] The sentence has been rephrased too (see https://github.com/DorianDepriester/MTEX2Gmsh/commit/256cec593c150e32978c4808cf9f111626beac48)
- [x] The typo has been fixed (see https://github.com/DorianDepriester/MTEX2Gmsh/commit/3a23e22a49c765400f12ecb1531394e2d2d26fe7)

**Comment by DorianDepriester** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02094/joss.02094/10.21105.joss.02094.pdf)
